src/app/services/tests/GameService.spec.ts

```diff
@@ -121,38 +94,54 @@ describe('GameService', () => {
-                    player: player.value,
-                },
-                typeGame: 'Quarto',
-                playerZero: UserMocks.CREATOR_MINIMAL_USER,
-                playerOne: UserMocks.OPPONENT_MINIMAL_USER,
-                turn: 2,
-                listMoves: [MOVE_1, MOVE_2],
-                request: Request.takeBackAsked(player),
-                result: MGPResult.UNACHIEVED.value,
-            });
-            await expectAsync(gameService.acceptTakeBack('configRoomId', part, player, [0, 1]))
-                .toBeRejectedWithError('Assertion failure: ' + error);
-            expect(ErrorLoggerService.logError).toHaveBeenCalledWith('Assertion failure', error);
-        }
-    }));
-    it('acceptConfig should delegate to ConfigRoomService and call startGameWithConfig', fakeAsync(async() => {
-        const configRoomService: ConfigRoomService = TestBed.inject(ConfigRoomService);
-        const configRoom: ConfigRoom = ConfigRoomMocks.WITH_PROPOSED_CONFIG;
-        spyOn(configRoomService, 'acceptConfig').and.resolveTo();
-        spyOn(partDAO, 'update').and.resolveTo();
+    describe('acceptConfig', () => {
+        it('should delegate to ConfigRoomService.acceptConfig', fakeAsync(async() => {
+            const configRoomService: ConfigRoomService = TestBed.inject(ConfigRoomService);
+            spyOn(configRoomService, 'acceptConfig').and.resolveTo();
+            spyOn(partDAO, 'update').and.resolveTo();
 
-        await gameService.acceptConfig('partId', configRoom);
+            // Given a config
+            const configRoom: ConfigRoom = ConfigRoomMocks.WITH_PROPOSED_CONFIG;
+            // When accepting it
+            await gameService.acceptConfig('partId', configRoom);
+            // Then acceptConfig should be called
+            expect(configRoomService.acceptConfig).toHaveBeenCalledOnceWith('partId');
+        }));
+        it('should can startGame with the accepter player as argument (Player.ZERO)', fakeAsync(async() => {
+            const configRoomService: ConfigRoomService = TestBed.inject(ConfigRoomService);
+            const gameEventService: GameEventService = TestBed.inject(GameEventService);
+            spyOn(configRoomService, 'acceptConfig').and.resolveTo();
+            spyOn(partDAO, 'update').and.resolveTo();
+            spyOn(gameEventService, 'startGame').and.resolveTo();
 
-        expect(configRoomService.acceptConfig).toHaveBeenCalledOnceWith('partId');
-    }));
+            // Given a config where we will start as player 0
+            const configRoom: ConfigRoom = {
+                ...ConfigRoomMocks.WITH_PROPOSED_CONFIG,
+                firstPlayer: FirstPlayer.CHOSEN_PLAYER.value,
+            };
+            // When accepting it
+            await gameService.acceptConfig('partId', configRoom);
+            // Then startGame is called with Player.ZERO
+            expect(gameEventService.startGame).toHaveBeenCalledWith('partId', Player.ZERO);
+        }));
```

Comment:
1. should call*? (110à
2. we will start as Player.ONE if, as creator, the config have firstPlayer = chosenPlayer, so misleading comment (117)
3. pas spécial que accept config soit appellé par le createur ?

Reply:
1. tellement
2. bawé, corrigé !
3. oui c'est misleading, c'est l'accepteur qui accepte, j'ai corrigé ce test ainsi que le suivant